doc/Requirements.md

```diff
@@ -0,0 +1,374 @@
+
+# Parameters
+
+A PF MUST be able to handle parameters.
+(Why: because the PF aims to abstract hardware and model it by parameters.)
+
+## Definitions
+
+hardware: system controlled by the PF. Not necessary material system.
+          This term was chosen because:
+ - historically the PF reference implementation was used to abstract hardware
+ - the subsystem term would arguably fit best is already used.
+(FIXME: choose subsystem instead, it is not used in fact.)
+
+## Value
+
+A parameter MUST have a value.
+(Why: because a parameter without value would not abstract any hardware.)
+
+A PF MUST support mutable parameters.
+(Why: To control the underlined hardware.)
+
+A PF MAY support immutable parameters, ie parameters which value is determined on start than read only.
+(Why: To permit hardware read only value reflection.)
+(Note: This is not implemented in the PF reference implementation.)
+
+This value MUST be gettable and settable (except if immutable).
+(Why: A parameter that can not be accessed is of no use.)
+
+A value MUST be convertible to a string.
+(Why: To display a parameter value to the human user for debuging purpose.)
+
+A bijection string to and from value SHOULD exist.
+(Why: To support serialization and deserialization of the parameter values.)
+
+### Data type
+
+#### Definition
+All parameters have a data type.
+A data type designates parameter invariants.
+A data type is the the meaning of the data and the way values of that type can be stored.
+
+
+#### Philosophy
+
+A data type defines the value properties:
+ - memory layout
+ - value constrains
+
+A value type is mostly used:
+ - pretty display a parameter (not just a as an array of bits)
+ - check for user error when setting it (out of bound, invalid...)
+ - offer a type safe api
+
+#### Requirements
+
+A PF SHOULD support the following types.
+If a type is chosen to be supported, it MUST respect all MUST clause,
+SHOULD respect all SHOULD clause, MAY respect all MAY clause of the type.
+(Why: All type are not necessary to use the PF. For example any parameter could
+      be represented as an array of char (string). But this would not permit to
+      check parameter validity (invariants) as well a pretty displaying the values.)
+
+Implementation MAY add add other API to access a parameter value.
+(Why: For example a C++ implementation may give access to a string as a std::string object.)
+
+##### Integers
+PF SHOULD support signed and unsigned integer parameters
+(Why: The reference implementation supports it.)
+
+PF MUST support integer with invariant size.
+(Why: It is common in C API to expect numbers to have a fixed maximum size.)
+
+The API to assess it MUST respect C abi:
+ - 2's complement for signed integer
+ - bits represent the power of two of their index.
+(Why: For easy access from C code.)
+
+Supported integer size SHOULD be at least 8, 16, 32 and 64 bits.
+(Why: The reference implementation supports it.)
```

Comment:
- You are right about 64 bit, they are not allowed in xsd.
- wrong, the code does not support 64 bit integers: see [IntegerParameterType.cpp](https://github.com/01org/parameter-framework/blob/master/parameter/IntegerParameterType.cpp) there are are int32_t cast literally everywhere.
 - I will remove `and 64`